roachtest: add a tag for the owner

tbg opened the pull request:
Automatically add an `owner-<X>` tag to each roachtest. This allows
listing all of the tests owned by a certain owner, for example

    ./bin/roachtest list tag:owner-kv

Furthermore a bit of piping can generate a regexp suitable for usage
in roachdash:

    ./bin/roachtest list tag:owner-kv | \
      grep -v 'skipped:' | \
      grep -Eo '^[^ ]+' | tr '\n' '|'

(You need to cut the trailing `|` and prefix with `.*` to make it work
in roachdash).

Ideally we would also add a Project field to each OwnerMetadata and
assign to that project upon creating an issue. Alas, the GitHub API does
not let you do that:

https://developer.github.com/v3/issues/#create-an-issue

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/45928)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> good idea.

> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45928#-:-M23mG6G9oqWHiCb5pTg:bbzbusj)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

tbg commented:
bors r=nvanbenschoten

andreimatei reviewed (approved):
LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/45928#-:-M787Xyg8z_QECcqy4mv:bb6ywmx)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andreimatei)



<!-- Sent from Reviewable.io -->

tbg committed:
roachtest: add a tag for the owner

Automatically add an `owner-<X>` tag to each roachtest. This allows
listing all of the tests owned by a certain owner, for example

    ./bin/roachtest list tag:owner-kv

Furthermore a bit of piping can generate a regexp suitable for usage
in roachdash:

    ./bin/roachtest list tag:owner-kv | \
      grep -v 'skipped:' | \
      grep -Eo '^[^ ]+' | tr '\n' '|'

(You need to cut the trailing `|` and prefix with `.*` to make it work
in roachdash).

Ideally we would also add a Project field to each OwnerMetadata and
assign to that project upon creating an issue. Alas, the GitHub API does
not let you do that:

https://developer.github.com/v3/issues/#create-an-issue

Release note: None

tbg commented:
bors r=andreimatei,nvanbenschoten

Fixed a bug - had to also make sure the default tag was there if the tag list was otherwise empty.

tbg commented:
bors r=andreimatei,nvanbenschoten

CI was broken
